mavlink - clarify all.xml rather than ardupilotmega (#173)
mavlink/qgc-dev-guide: en/custom_build/mavlink.md

# MAVLink Customisation

QGC communicates with flight stacks using [MAVLink](https://mavlink.io/en/), a very lightweight messaging protocol that has been designed for the drone ecosystem.

> QGC includes the [all.xml](https://mavlink.io/en/messages/all.html) dialect by default from QGC 4.2.8, which includes all other dialects in the [mavlink/mavlink](https://github.com/mavlink/mavlink/tree/master/message_definitions/v1.0 ) repository, and allows it to communicate with both PX4 and Ardupilot.

Author: @DonLakeFlyer Is this the right version of QGC? Otherwise is this OK?

I assume that ArduPilot might use development.xml too, but don't state that.

Reviewer: I'm assuming these two commit represent the change that @DonLakeFlyer mention: 
- https://github.com/mavlink/qgroundcontrol/commit/c716fbb28383df9984f9efd84c70e6fb39390d28
- https://github.com/mavlink/qgroundcontrol/commit/cb39702cc6ba729333d139c1bd6047255de11d76
I think it was accidental that the v4.2.8 worked with the `all.xml` since it include the `ArduPilot.xml`.

So I would say something like this:
```suggestion
QGC includes the [all.xml](https://mavlink.io/en/messages/all.html) dialect by default. The `all.xml` includes all other dialects in the [mavlink/mavlink](https://github.com/mavlink/mavlink/tree/master/message_definitions/v1.0 ) repository, and allows it to communicate with both PX4 and Ardupilot. Previous versions of QGC, since v4.2.8, used the `ArduPilotMega.xml` dialect.
```

That way, we explained that the previous version used the `ArduPilotMega.xml` and that the future release will use the `all.xml`. Feel free to rephrase it.

> QGC includes the [all.xml](https://mavlink.io/en/messages/all.html) dialect by default. The `all.xml` includes all other dialects in the [mavlink/mavlink](https://github.com/mavlink/mavlink/tree/master/message_definitions/v1.0 ) repository, and allows it to communicate with both PX4 and Ardupilot. Previous versions of QGC, since v4.2.8, used the `ArduPilotMega.xml` dialect.

Author: ```suggestion
QGC includes the [all.xml](https://mavlink.io/en/messages/all.html) dialect by default. The `all.xml` includes all other dialects in the [mavlink/mavlink](https://github.com/mavlink/mavlink/tree/master/message_definitions/v1.0 ) repository, and allows it to communicate with both PX4 and Ardupilot. 
Previous versions of QGC (v4.2.8 and earlier), used the `ArduPilotMega.xml` dialect.
```

In order to add support for a new set of messages you should add them to [development.xml](https://mavlink.io/en/messages/development.html), [ArduPilotMega.xml](https://mavlink.io/en/messages/ardupilotmega.html), or [common.xml](https://mavlink.io/en/messages/common.html) (for PX4), or fork *QGroundControl* and include your own dialect.

To modify the version of MAVLink used by QGC:

- Replace the pre-build C library at [/qgroundcontrol/libs/mavlink/include/mavlink](https://github.com/mavlink/qgroundcontrol/tree/master/libs/mavlink/include/mavlink).
  - By default this is a submodule importing https://github.com/mavlink/c_library_v2
  - You can change the submodule, or [build your own libraries](https://mavlink.io/en/getting_started/generate_libraries.html) using the MAVLink toolchain.
- You can change the whole dialect used by setting it in [`MAVLINK_CONF`](https://github.com/mavlink/qgroundcontrol/blob/master/QGCExternalLibs.pri#L52) when running *qmake*.

> Note that earlier versions of QGC used [ArduPilotMega.xml](https://mavlink.io/en/messages/ardupilotmega.html) instead of all.xml.
> If you are working on these versions then add definitions to common.xml or ArduPilotMega.xml (but not development.xml).

Author: No longer necessary
```suggestion
```
